### PR TITLE
goaws : fake sqs/sns server

### PR DESCRIPTION
**Pull request by deeptibaghel**
PR to test test goaws fake sqs/sns server

### CODE REVIEW

File: config/local.json-dist [hunk 1]
```diff
@@ -1,6 +1,16 @@
 {
-  "env": "dev",
+  "env": "development",
   "log": {
     "level": "debug"
+  },
+  "basket":{
+    "sqs": {
+       "region": {
+         "default": "local"
```
Comment (vladikoff) [inline config/local.json-dist]: there should be no `default` property, just the value: http://v14d.com/i/5aba99c75c39c.jpg

**Comment by vladikoff** [comment]
@deeptibaghel as part of fxa-local-dev we also need to start this file https://github.com/mozilla/fxa-basket-proxy/blob/master/bin/basket-event-handler.js

if sign up and verify you should output in that process, such as:


> {"Timestamp":1522178802031000000,"Logger":"fxa-basket-proxy","Type":"events.handleEvent","Severity":6,"Pid":99559,"EnvVersion":"2.0","Fields":{"email":"x@x.com","uid":"b2b64025bc064bf5baeb30b82c41cd9c","locale":"en-US,en;q=0.9,pt;q=0.8,af;q=0.7","marketingOptIn":true,"ts":1522178801.944,"metricsContext":"{\"time\":1522178801944,\"device_id\":\"76191b88d8334c8591d27b50dc67add8\",\"flow_id\":\"39f51ca0388cf205fb99b7a7d6310cbcc1f379996c6da478766a4433dd0ecc94\",\"flow_time\":14052,\"flowBeginTime\":1522178787892,\"flowCompleteSignal\":\"account.verified\",\"flowType\":\"registration\"}","event":"verified"}}
{"Timestamp":1522178802067000000,"Logger":"fxa-basket-proxy","Type":"events.event.request.error.http","Severity":2,"Pid":99559,"EnvVersion":"2.0","Fields":{"email":"x@x.com","uid":"b2b64025bc064bf5baeb30b82c41cd9c","locale":"en-US,en;q=0.9,pt;q=0.8,af;q=0.7","marketingOptIn":true,"ts":1522178801.944,"metricsContext":"{\"time\":1522178801944,\"device_id\":\"76191b88d8334c8591d27b50dc67add8\",\"flow_id\":\"39f51ca0388cf205fb99b7a7d6310cbcc1f379996c6da478766a4433dd0ecc94\",\"flow_time\":14052,\"flowBeginTime\":1522178787892,\"flowCompleteSignal\":\"account.verified\",\"flowType\":\"registration\"}","event":"verified","endpoint":"/fxa-register/","status":404,"body":"<!DOCTYPE html>\n<html lang=\"en\">\n<head>\n<meta charset=\"utf-8\">\n<title>Error</title>\n</head>\n<body>\n<pre>Cannot POST /fxa-register/</pre>\n</body>\n</html>\n"}}
{"Timestamp":1522178802081000000,"Logger":"fxa-basket-proxy","Type":"events.event.request","Severity":6,"Pid":99559,"EnvVersion":"2.0","Fields":{"email":"x@x.com","uid":"b2b64025bc064bf5baeb30b82c41cd9c","locale":"en-US,en;q=0.9,pt;q=0.8,af;q=0.7","marketingOptIn":true,"ts":1522178801.944,"metricsContext":"{\"time\":1522178801944,\"device_id\":\"76191b88d8334c8591d27b50dc67add8\",\"flow_id\":\"39f51ca0388cf205fb99b7a7d6310cbcc1f379996c6da478766a4433dd0ecc94\",\"flow_time\":14052,\"flowBeginTime\":1522178787892,\"flowCompleteSignal\":\"account.verified\",\"flowType\":\"registration\"}","event":"verified","endpoint":"/subscribe/","status":200,"body":"{\"status\":\"ok\"}"}}

File: lib/config.js [hunk 2]
```diff
@@ -147,6 +147,15 @@ if (! process.env.NODE_ENV) {
   process.env.NODE_ENV = conf.get('env');
 }
 
+if(conf.get('env')  === 'development'){
```
Comment (vladikoff) [inline lib/config.js]: see 2 comments for this from https://github.com/mozilla/fxa-auth-server/pull/2369#pullrequestreview-107829584

**Comment by vladikoff** [comment]
@deeptibaghel I have this working with:

```diff
diff --git a/lib/events/sqs.js b/lib/events/sqs.js
index 2d3d0a5..8d2382c 100644
--- a/lib/events/sqs.js
+++ b/lib/events/sqs.js
@@ -65,8 +65,7 @@ SQSReceiver.prototype.fetch = function (url) {
         var msg = data.Messages[i];
         var deleteFromQueue = deleteMessage.bind(this, msg);
         try {
-          var body = JSON.parse(msg.Body);
-          var message = JSON.parse(body.Message);
+          var message = JSON.parse(msg.Body);
           message.del = deleteFromQueue;
           this.emit('data', message);
         }
```

but @rfk tells me we are missing a layer of fake SNS/SQS somewhere 🔍 .

I shall keep looking

**Comment by deeptibaghel** [comment]
@vladikoff so those communication tests pass with this change ?

**Comment by vladikoff** [comment]
Currently the server won't subscribe `restmail.net` addresses, we need to make that configurable and disable that check in `development`

**Comment by vladikoff** [comment]
The check is right here: https://github.com/mozilla/fxa-basket-proxy/blob/39d19ee6d3973cfd4a9de66b3699a8ffa3f43044/lib/events/index.js#L182

**Comment by rfk** [comment]
> Currently the server won't subscribe restmail.net addresses, we need to make that configurable

+1 to making this configurable, similar to various email-regex checks in the auth-server.  FYI basket itself also has a similar check, but it's configurable in that case:

  https://github.com/mozmeao/basket/blob/6751a9626c0442e400a4368ae38112c85aad325c/basket/base/utils.py#L4

**Comment by vladikoff** [comment]
@deeptibaghel see this example: https://github.com/mozilla/fxa-auth-server/commit/d62995ea5089ada6452066d2d7528922bde9db69

**Comment by deeptibaghel** [comment]
@vladikoff 
I think it's not reaching handleEvent event. I am seeing this message though: 
{"Timestamp":1530890254079000000,"Logger":"fxa-basket-proxy","Type":"verify.auth.invalid-scope","Severity":2,"Pid":1033,"EnvVersion":"2.0","Fields":{"user":"c4b4d5481cbb45dd9c86fc9422c382d9","client_id":"98e6508e88680e1a","scope":"[\"basket\",\"profile:email\"]"}}

**Pull request closed without merging**